src/common/scratch_buffer.hpp

```diff
@@ -1,18 +1,24 @@
-// scratch_buffer.h
-
-#include <inttypes.h>
-
+///
+/// \file
 /// Single place for definition of large RAM space for various tasks.
-/// Currently, there's no locking mechanism so before you use it, make sure
-/// there's nobody else who uses it.
 ///
-/// This is used by:
-/// - PNG decoder
-/// - QR code generator
+/// The scratch buffer isn't safe-to-use outside of the FreeRTOS context.
 ///
 /// TODO: use it for
 /// - SHA256 encoding
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
This suggest it is C header, so name of the file is supposed to be scratch_buffer.h according to our code style. *.hpp is reserved for header incompatible with C.

Reply:
What? I wasn't aware I added the file 🙈

Reply:
Ah, I see. I made it usable from C...

Reply:
Fixed by removing all changes to the scratch_buffer. They weren't need in the last version this PR.